src/main/java/org/cyclonedx/util/DependencySerializer.java

```diff
@@ -49,32 +49,57 @@ public DependencySerializer(final Class t) {
   public void serialize(
       final List<Dependency> depList, final JsonGenerator generator, final SerializerProvider provider)
   {
-    final ToXmlGenerator toXmlGenerator = (ToXmlGenerator) generator;
-    final XMLStreamWriter staxWriter = toXmlGenerator.getStaxWriter();
-    if (useNamespace) {
-      try {
-        if (depList != null && !depList.isEmpty()) {
-          staxWriter.writeStartElement(NAMESPACE_PREFIX, "dependencies", NAMESPACE_URI);
-          for (Dependency d : depList) {
-            writeDependency(d, staxWriter);
+    if (generator instanceof ToXmlGenerator) {
+      final ToXmlGenerator toXmlGenerator = (ToXmlGenerator) generator;
+      final XMLStreamWriter staxWriter = toXmlGenerator.getStaxWriter();
+      if (useNamespace) {
+        try {
+          if (depList != null && !depList.isEmpty()) {
+            staxWriter.writeStartElement(NAMESPACE_PREFIX, "dependencies", NAMESPACE_URI);
+            for (Dependency d : depList) {
+              writeDependency(d, staxWriter);
+            }
           }
         }
+        catch (XMLStreamException e) {
+          e.printStackTrace();
+        }
       }
-      catch (XMLStreamException e) {
-        e.printStackTrace();
+      else {
+        try {
+          if (depList != null && !depList.isEmpty()) {
+            staxWriter.writeStartElement("dependencies");
+            for (Dependency dep : depList) {
+              writeNormalDependency(dep, staxWriter);
+            }
+            staxWriter.writeEndElement();
+          }
+        }
+        catch (XMLStreamException e) {
+          e.printStackTrace();
+        }
```

Comment:
Those two parts look like the same, something like the main PR can be done to reduce duplicated one, having only one method and do the check there

Reply:
In: 
9f6c7d2